packages/core/src/fst/index.ts

```diff
@@ -0,0 +1,26 @@
+// https://en.wikipedia.org/wiki/Finite-state_transducer
+
+import type { State, StateMachine } from '..';
+
+export { toActor as fromActor } from './actor';
+export { fromMachine } from './machine';
+export { fromReducer } from './reducer';
+
+export type FSTTransition<TState, TEvent, TOutput> = (
+  state: TState,
+  input: TEvent
+) => [TState] | [TState, TOutput];
```

Comment:
This return type won't play nicely with the consuming side of things. This code snippet should be valid:
```js
type FSTTransition<TState, TEvent, TOutput> = (
  state: TState,
  input: TEvent
) => [TState] | [TState, TOutput];

type MyState = { count: number }
type MyEvent = { type: 'INC' }
type MyOutput = { type: 'alert' }

declare const transition: FSTTransition<{ count: number }, MyEvent, MyOutput>
declare const handleOutput: (output: MyOutput) => void

let [state, output] = transition({ count: 0 }, { type: "INC" })
handleOutput(output) // oops, error since output is of type `MyOutput | undefined`
```

Unless this is the intention but then the return type of `[TState, TOutput | undefined]` would be more explicit about it.